server/events/project_command_runner.go

```diff
@@ -213,8 +215,14 @@ func (p *DefaultProjectCommandRunner) doApply(ctx models.ProjectCommandContext)
 	}
 	// todo: this class shouldn't know about the server-side approval requirement.
 	// Instead the project_command_builder should figure this out and store this information in the ctx. # refactor
-	if p.RequireApprovalOverride {
-		applyRequirements = []string{raw.ApprovedApplyRequirement}
+	if p.RequireMergeableOverride || p.RequireApprovalOverride {
```

Comment:
Let's move this up and combine it with our check of `ctx.ProjectConfig`. Also if users specify both flags then we should use both requirements:

```go
	// Figure out what our apply requirements are.
	var applyRequirements []string
	if p.RequireApprovalOverride || p.RequireMergeableOverride {
		// If any server flags are set, they override project config.
		if p.RequireApprovalOverride {
			applyRequirements = append(applyRequirements, raw.ApprovedApplyRequirement)
		}
		if p.RequireMergeableOverride {
			applyRequirements = append(applyRequirements, raw.MergeableApplyRequirement)
		}
	} else if ctx.ProjectConfig != nil {
		// Else we use the project config if it's set.
		applyRequirements = ctx.ProjectConfig.ApplyRequirements
	}
```

Reply:
Got it, will do.